objects/boat/Boat.gd

```diff
@@ -0,0 +1,8 @@
+class_name Boat
+extends StaticBody2D
+
+var _can_enter = false
+
+func _on_enter():
+	if _can_enter:
+		SceneTransition.change_scene("res://scenes/map/Map.tscn")
```

Comment:
It would be easier to evolve if the boat only emit a signal and the level listen to the signal to change level.